File: src/main/resources/templates/dart/solution/lib/policy.dart
import 'bubble_sort.dart';
import 'context.dart';
import 'merge_sort.dart';

class Policy {
  static const int mergeSortThreshold = 10;

  final Context context;

  Policy(this.context);

  /// Chooses a strategy depending on the number of date objects.
  void configure() {
    final dates = context.dates;
    if (dates == null) {
      throw StateError(
          'dates of Context has to be set before configure() can be called');
    }

    if (dates.length >= mergeSortThreshold) {
      context.sortAlgorithm = MergeSort();
    } else {
      context.sortAlgorithm = BubbleSort();
    }
  }
}

File: src/main/resources/templates/go/solution/policy.go
package ${packageName}

type Policy struct {
	context *Context
}

const datesSizeThreshold = 10

func NewPolicy(context *Context) *Policy {
	return &Policy{context}
}

// Configure chooses a strategy depending on the number of date objects.
func (p *Policy) Configure() {
	if len(p.context.GetDates()) > datesSizeThreshold {
		p.context.SetSortAlgorithm(NewMergeSort())
	} else {
		p.context.SetSortAlgorithm(NewBubbleSort())
	}
}
